elcometer456_block.py

```diff
@@ -17,23 +18,38 @@ class Elcometer456(Block):
     def __init__(self):
         super().__init__()
         self._serial = None
-        self._thread = None
+        self._connect_thread = None
+        self._read_thread = None
         self._stopping = False
 
-    def configure(self, context):
-        super().configure(context)
-        self._serial = serial.Serial(self.port(), self.baudrate(), timeout=self.timeout())
-
     def start(self):
         super().start()
-        self._thread = spawn(self._read_gage)
+        self._connect_thread = spawn(self._connect_gage)
 
     def stop(self):
         self._stopping = True
-        self._serial.close()
-        self._thread.join(1)
+        if self._connect_thread:
+            self._connect_thread.join(1)
+        if self._serial:
+            self._serial.close()
+        if self._read_thread:
+            self._read_thread.join(1)
         super().stop()
 
+    def _connect_gage(self):
+        while True:
+            if self._stopping:
+                return
+            try:
+                self._serial = serial.Serial(self.port(), self.baudrate(), timeout=self.timeout())
+                self.logger.info('Pairing Successful')
+            except:
+                self.logger.warning('Pairing attempt failed', exc_info = False)
+                sleep(10)
+                continue
+            break
+        self._read_thread = spawn(self._read_gage)
```

Comment:
I think we could simplify this and remove the `spawn` here. It was necessary before because it was in `start` and start needs to spawn long running methods so that it can return before the service times out. _connect_gage is already in it's own thread so this may be unnecessary.